File: Data Structures & Algorithms/minimum-stack/submission-0.java
class MinStack {
    Stack<Integer> ms = new Stack<>();
    Stack<Integer> s = new Stack<>();

    public MinStack() {
        
    }
    
    public void push(int val) {
        s.push(val);
        if(ms.isEmpty()){
            ms.push(val);
        }
        else {
            if(ms.peek() >= val){
                ms.push(val);
            }
        }
    }
    
    public void pop() {
        int a = s.pop();
        if(a == ms.peek()){
            ms.pop();
        }
    }
    
    public int top() {
        return s.peek();
    }
    
    public int getMin() {
        return ms.peek();
    }
}

File: Data Structures & Algorithms/implement-queue-using-stacks/submission-0.cpp
class MyQueue {
public:
    stack<int> s1, s2;
    MyQueue() {
        
    }
    
    void push(int x) {

            while(!s1.empty()){
                int top = s1.top();
                s2.push(top);
                s1.pop();
            }
            s1.push(x);
            while(!s2.empty()){
                int top = s2.top();
                s1.push(top);
                s2.pop();
            }
        }
    
    
    int pop() {
        int top = s1.top();
        s1.pop();
        return top;
    }
    
    int peek() {
        return s1.top();
    }
    
    bool empty() {
        return s1.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */